.github/workflows/maturin.yml

```diff
@@ -0,0 +1,76 @@
+# This file is autogenerated by maturin v1.7.4 and manually modified by ia0.
+name: Maturin
+
+on:
+  pull_request:
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    runs-on: ${{ matrix.platform.runner }}
+    strategy:
+      matrix:
+        platform:
+          - runner: ubuntu-latest
+            target: x86_64
+          - runner: windows-latest
+            target: x64
+          - runner: macos-14
+            target: aarch64
+    steps:
+      - uses: actions/checkout@v4
+      - if: matrix.platform.runner == 'ubuntu-latest'
+        uses: actions/cache@0c45773b623bea8c8e75f6c82b208c3cf94ea4f9 # v4.0.2
+        with:
+          path: rust/onnx/runtime/build/Linux
+          key: maturin-${{ matrix.platform.target }}-${{ hashFiles('rust/onnx/build.sh') }}
+      - if: matrix.platform.runner == 'ubuntu-latest'
+        name: Build wheels
+        uses: PyO3/maturin-action@v1
+        with:
+          target: ${{ matrix.platform.target }}
+          args: --release --out=../dist
+          before-script-linux: "${{ github.workspace }}/rust/onnx/maturin.sh"
+          manylinux: 2_28
+          working-directory: python
+      - if: matrix.platform.runner != 'ubuntu-latest'
+        name: Build wheels
+        uses: PyO3/maturin-action@v1
+        with:
+          target: ${{ matrix.platform.target }}
+          args: --release --out=../dist
+          working-directory: python
+      - name: Rewrite wheels
+        run: true # TODO(reyammer): Fix wheels version.
```

Comment:
> ack. I can patch the version number; anything else I need to take care of?

I don't know. This would need a wheels inspection. I looked at the Python library version and the Rust CLI version. Both are good.

> BTW, any reason to not do what we did before, that is: patch the version number before running the build? patching a wheel is doable, but I'm worried about potential side effects that we discover later on...

Good point, this is actually much better. You still need to inspect the wheels though to see if there are any other problems. But at least the process is not too hacky (the Rust version rewrite is at least well defined).

Reply:
ack, I'll take a pass at this in a separate PR.